### PR TITLE
Docker unattended-upgrades

### PR DESCRIPTION
**Pull request by msladek**
I run Debian instances with [unattended-upgrades](https://wiki.debian.org/UnattendedUpgrades) installed. When applying this role to some of those servers the packages from the Docker repo don't get automatically updated since the package control `Unattended-Upgrade::Origins-Pattern` for Docker is missing.

This PR adds this configuration for Debian systems during rootful installation in case the package `unattended-upgrades` is installed. It does nothing otherwise.

### CODE REVIEW

**Comment by konstruktoid** [comment]
I like the idea and `[unattended-upgrades](https://wiki.debian.org/UnattendedUpgrades)` but it needs to be a variable with default `false`, if people dont use `live-restore` alot of systems will be crashing down.

**Comment by msladek** [comment]
@konstruktoid I've added a variable to make sure it isn't accidentality configured.

**Review by konstruktoid — CHANGES_REQUESTED** [review]
So sorry for the delay!
There are a couple of improvements, and it needs `become` to write the config file.

```diff
diff --git a/molecule/default/molecule.yml b/molecule/default/molecule.yml
index 09eece4..10f4e11 100644
--- a/molecule/default/molecule.yml
+++ b/molecule/default/molecule.yml
@@ -62,6 +62,7 @@ provisioner:
         docker_allow_privileged_ports: false
         docker_rootful: true
         docker_rootful_enabled: false
+        docker_unattended_upgrades: true
         docker_user: dockeruser
         docker_user_bashrc: false
         docker_compose: false
diff --git a/tasks/docker_install_rootful.yml b/tasks/docker_install_rootful.yml
index 5b9b503..32f7edb 100644
--- a/tasks/docker_install_rootful.yml
+++ b/tasks/docker_install_rootful.yml
@@ -29,15 +29,18 @@
         manager: auto

     - name: Unattended upgrades for Docker
+      become: true
       ansible.builtin.lineinfile:
         path: /etc/apt/apt.conf.d/55unattended-upgrades-docker
-        line: 'Unattended-Upgrade::Origins-Pattern { "o=Docker,a=${distro_codename}"; };'
+        line: Unattended-Upgrade::Origins-Pattern { "o=Docker,a=${distro_codename}"; };
         state: present
         create: true
         owner: root
         group: root
         mode: "0644"
-      when: docker_unattended_upgrades | bool and 'unattended-upgrades' in ansible_facts.packages
+      when:
+        - docker_unattended_upgrades | bool
+        - "'unattended-upgrades' in ansible_facts.packages"

     - name: Run apt update
       become: true
```

**Comment by msladek** [comment]
@konstruktoid the patch is applied, sorry for hiatus

File: tasks/docker_install_rootful.yml [hunk 1]
```diff
@@ -24,6 +24,24 @@
...
+      become: true
+      ansible.builtin.lineinfile:
+        path: /etc/apt/apt.conf.d/55unattended-upgrades-docker
+        line: Unattended-Upgrade::Origins-Pattern { "o=Docker,a=${distro_codename}"; };
+        state: present
+        create: true
+        owner: root
+        group: root
+        mode: "0644"
+        when:
```
Comment (konstruktoid) [inline tasks/docker_install_rootful.yml:41]: wrong identation

**Review by konstruktoid — CHANGES_REQUESTED** [review]

**Comment by konstruktoid** [comment]
> @konstruktoid the patch is applied, sorry for hiatus

no worries whatsoever, thanks for updating it

**Review by konstruktoid — APPROVED** [review]

**Comment by konstruktoid** [comment]
So sorry for the delay, and thank you for the PR

**Pull request merged**